var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -115,45 +134,70 @@ def url_for_version(self, version):
     depends_on('python', when='+python')
     depends_on('mpi', when='+user-lb')
+    depends_on('mpi', when='+latboltz')
     depends_on('mpi', when='+user-h5md')
     depends_on('hdf5', when='+user-h5md')
+    depends_on('mpi', when='+h5md')
+    depends_on('hdf5', when='+h5md')
     depends_on('jpeg', when='+jpeg')
     depends_on('kim-api', when='+kim')
     depends_on('libpng', when='+png')
     depends_on('ffmpeg', when='+ffmpeg')
     depends_on('kokkos+deprecated_code+shared@3.0.00', when='@20200303+kokkos')
     depends_on('kokkos+shared@3.1:', when='@20200505:+kokkos')
     depends_on('adios2', when='+user-adios')
+    depends_on('adios2', when='+adios')
     depends_on('plumed', when='+user-plumed')
+    depends_on('plumed', when='+plumed')
     depends_on('eigen@3:', when='+user-smd')
+    depends_on('eigen@3:', when='+machdyn')
 
     conflicts('+cuda', when='+opencl')
     conflicts('+body', when='+poems@:20180628')
     conflicts('+latte', when='@:20170921')
     conflicts('+python', when='~lib')
     conflicts('+qeq', when='~manybody')
     conflicts('+user-atc', when='~manybody')
+    conflicts('+atc', when='~manybody')
     conflicts('+user-misc', when='~manybody')
     conflicts('+user-phonon', when='~kspace')
-    conflicts('+user-misc', when='~manybody')
+    conflicts('+phonon', when='~kspace')
     conflicts('%gcc@9:', when='@:20200303+openmp')
     conflicts('+kokkos', when='@:20200227')
     conflicts(
-        '+meam', when='@20181212:',
-        msg='+meam was removed after @20181212, use +user-meamc instead')
+        '+meam', when='@20190104:20210527',
+        msg='+meam was removed from version @20190104 to version @20210527, ' +
+            'use +user-meamc instead')
     conflicts(
-        '+user-meamc', when='@:20181212',
-        msg='+user-meamc only added @20181212, use +meam instead')
+        '+user-meamc', when='@:20181212,20210702:',
+        msg='+user-meamc only supported after version @20181212 and before ' +
+            'version @20210702, use +meam instead')
     conflicts(
         '+user-reaction', when='@:20200303',
         msg='+user-reaction only supported for version 20200505 and later')
     conflicts('+mliap', when='~snap')
+    conflicts('+ml-iap', when='~ml-snap')
     conflicts(
         '+user-adios +mpi', when='^adios2~mpi',
         msg='With +user-adios, mpi setting for adios2 and lammps must be the same')
     conflicts(
         '+user-adios ~mpi', when='^adios2+mpi',
         msg='With +user-adios, mpi setting for adios2 and lammps must be the same')
+    conflicts(
+        '+adios +mpi', when='^adios2~mpi',
+        msg='With +adios, mpi setting for adios2 and lammps must be the same')
+    conflicts(
+        '+adios ~mpi', when='^adios2+mpi',
+        msg='With +adios, mpi setting for adios2 and lammps must be the same')
+
+    for old_pkg, new_pkg in renamed_packages_20210702.items():
+        conflicts(
+            f'+{old_pkg}', when='@20210702:',
```

Comment:
F-strings were introduced in Python 3.6, but Spack supports Python 2.6-2.7 and 3.5. That's why the validate tests are failing. You'll have to use normal formatting like `'+{0}'.format(old_pkg)`

Reply:
Ping @RemiLacroix-IDRIS

Reply:
Sorry about the delay, I was kept busy with other things.

I understand the problem and will fix it shortly.